exporters/prometheus/exporter.go

```diff
@@ -181,6 +190,10 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 		ch <- c.targetInfo
 	}
 
+	if c.resourceAttributesFilter != nil {
+		c.resourceAttributes(metrics.Resource)
```

Comment:
Why does `c.resourceAttributes` store the resource attributes as `c.resourceKeyVals`? Why not just return a `keyVals` value from that method?

Reply:
it could return it, i just figure it would save an allocation by using r.resourceAttributes

Reply:
We've assumed resource is constant for a Collector thus far (e.g. for target_info).  In theory, someone using the prometheus exporter without the SDK could pass different resources, but IMO that is an antipattern

Reply:
Can we unify the `c.resourceAttributesFilter != nil` and `len(c.resourceKeyVals.keys) > 0` checks to be in the same method?

The current "straddling" layout seems to have lead to my confusion (:facepalm:).

Reply:
Moved the check outside the method, please take a look. Note that we could probably optimize the code a bit if we used a pointer to `keyVals`. Currently, in the case of someone setting a filter that returns no attributes, we'd always try to add filtered attributes, but i suspect this is an edge case or caused by misconfiguration.